pkg/kubelet/prober/prober_manager_test.go

```diff
@@ -130,6 +130,47 @@ func TestAddRemovePods(t *testing.T) {
 	if err := expectProbes(m, nil); err != nil {
 		t.Error(err)
 	}
+
+	// Adding duplicate probes, should not add probe worker
+	probeTypeStr2Enum := map[string]probeType{
+		"liveness":  liveness,
+		"readiness": readiness,
+		"startup":   startup,
+	}
+	probeType2Pods := make(map[string]*v1.Pod, 0)
+	for _, probeType := range []string{"liveness", "readiness", "startup"} {
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				UID: types.UID("dup_" + probeType + "_probe_pod"),
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{{Name: probeType}, {Name: probeType}},
+			},
+		}
+		switch probeType {
+		case "liveness":
+			pod.Spec.Containers[0].LivenessProbe = defaultProbe
+			pod.Spec.Containers[1].LivenessProbe = defaultProbe
+		case "readiness":
+			pod.Spec.Containers[0].ReadinessProbe = defaultProbe
+			pod.Spec.Containers[1].ReadinessProbe = defaultProbe
+		case "startup":
+			pod.Spec.Containers[0].StartupProbe = defaultProbe
+			pod.Spec.Containers[1].StartupProbe = defaultProbe
+		default:
+			t.Fatalf("unimplemented probe type %s", probeType)
+		}
+		probeType2Pods[probeType] = pod
+	}
+	for _, probeType := range []string{"liveness", "readiness", "startup"} {
```

Comment:
It doesn't look like this loop and probetype2Pods are needed. You can do this in one loop.